File: C++/2021/1stOct_mihir1739.cpp
#include<iostream>
#include<cmath>
using namespace std;
bool is_prime(long n)
{
    if (n == 1)
    {
        return true;
    }
    bool prime = true;
    int r = sqrt(n)+1;
    for (int i = 2; i <r; i++)
    {
        if (n % i == 0)
        {
            return false;
        }
    }
    return prime;
}

void fibbonacci(int n)
{
    long long a = 1;
    long long b = 1;
    int curr = 2;
    cout << a << ' ' << b <<' ';
    while(curr < n)
    {
        long long c = a + b;
        if (is_prime(c) || c % 5 == 0)
        {
            cout << '0' << ' ';
        }
        else
        {
            cout << c << ' ';
        }
        a = b;
        b = c;
        curr++;
    }
}
int main(){
    int n;
    cin >> n;
    fibbonacci(n);
    return 0;
}

File: Python/2021/1stOct_mihir1739.py
import math as m
def is_prime(n):
    r = int(m.sqrt(n))+1
    prime = True
    for i in range(2,r+1):
        if(n % i == 0):
            return False
    
    return prime

def fibonacci(o):
    a = 1
    b = 1
    print(a,end=' ')
    print(b,end=' ')
    curr = 2
    while(curr<o):
        c = a + b
        if(is_prime(c) or c % 5 == 0):
            print('0',end=' ')
        else:
            print(c,end=' ')
        a = b
        b = c
        curr += 1

g = int(input())
fibonacci(g)